sdk/iot/core/inc/az_iot_core.h

```diff
@@ -58,56 +62,49 @@ typedef enum
   AZ_IOT_STATUS_TIMEOUT = 504,
 } az_iot_status;
 
-/**
- * @brief Get the #az_iot_status from an int.
- *
- * @param[in] status_int The int with the status number.
- * @param[out] status The #az_iot_status* with the status enum.
- * @return The #az_result with the result of the get operation.
- *  @retval #AZ_OK If the int is an #az_iot_status enum. `status` will be set to the according
- * enum.
- *  @retval #AZ_ERROR_ITEM_NOT_FOUND If the int is NOT an #az_iot_status enum.
- */
-
-AZ_NODISCARD az_result az_iot_get_status_from_uint32(uint32_t status_int, az_iot_status* status);
-
 /**
  * @brief Checks if the status indicates a successful operation.
  *
  * @param[in] status The #az_iot_status to verify.
- * @return True if the status indicates success. False otherwise.
+ * @return `true` if the status indicates success. `false` otherwise.
  */
-AZ_NODISCARD bool az_iot_is_success_status(az_iot_status status);
+AZ_INLINE bool az_iot_is_success_status(az_iot_status status)
+{
+  return status < AZ_IOT_STATUS_BAD_REQUEST;
+}
 
 /**
  * @brief Checks if the status indicates a retriable error occurred during the
  *        operation.
  *
  * @param[in] status The #az_iot_status to verify.
- * @return True if the operation should be retried. False otherwise.
+ * @return `true` if the operation should be retried. `false` otherwise.
  */
-AZ_NODISCARD bool az_iot_is_retriable_status(az_iot_status status);
+AZ_INLINE bool az_iot_is_retriable_status(az_iot_status status)
+{
+  return ((status == AZ_IOT_STATUS_THROTTLED) || (status == AZ_IOT_STATUS_SERVER_ERROR));
+}
 
 /**
  * @brief Calculates the recommended delay before retrying an operation that failed.
  *
  * @param[in] operation_msec The time it took, in milliseconds, to perform the operation that
  *                           failed.
  * @param[in] attempt The number of failed retry attempts.
- * @param[in] retry_delay_msec The minimum time, in milliseconds, to wait before a retry.
+ * @param[in] min_retry_delay_msec The minimum time, in milliseconds, to wait before a retry.
  * @param[in] max_retry_delay_msec The maximum time, in milliseconds, to wait before a retry.
  * @param[in] random_msec A random value between 0 and the maximum allowed jitter, in milliseconds.
  * @return The recommended delay in milliseconds.
  */
 AZ_NODISCARD int32_t az_iot_retry_calc_delay(
     int32_t operation_msec,
     int16_t attempt,
-    int32_t retry_delay_msec,
+    int32_t min_retry_delay_msec,
     int32_t max_retry_delay_msec,
     int32_t random_msec);
 
 /**
- * @brief az_span_token is a string tokenizer for az_span.
+ * @brief _az_span_token is a string tokenizer for az_span.
```

Comment:
Remove method name from doc to avoid redundant info and having to make changes in multiple places in the future.

```suggestion
 * @brief String tokenizer for #az_span.
```

Reply:
Changed this and the one below.